### PR TITLE
Fix warnings in product-build

### PR DESCRIPTION
**Pull request by himasha**
## Approach
Included missing versions in dependencies.
Update pom parent path where applicable.
Remove duplicated dependencies.

### CODE REVIEW

File: samples/webapps/jaxrs_basic/pom.xml [hunk 1]
```diff
@@ -39,13 +39,15 @@
             </plugin>
             <plugin>
                 <artifactId>maven-war-plugin</artifactId>
+                <version>3.1.0</version>
                 <configuration>
```
Comment (a5anka) [inline samples/webapps/jaxrs_basic/pom.xml]: Shouldn't we inherit these versions from parent pom?

Comment (himasha) [reply samples/webapps/jaxrs_basic/pom.xml]: This plugin is not used in any other pom, so does it make sense to add it to parent pom?

Comment (wggihan) [reply samples/webapps/jaxrs_basic/pom.xml]: Just add version in the root pom and refer in here.

Comment (himasha) [reply samples/webapps/jaxrs_basic/pom.xml]: Fixed with 
15df4ec

File: samples/business-process-samples/product/bpmn-sample-jars/pom.xml [hunk 2]
```diff
@@ -28,7 +28,7 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-eclipse-plugin</artifactId>
-                <inherited>true</inherited>
+                <version>2.10</version>
```
Comment (chanikag) [inline samples/business-process-samples/product/bpmn-sample-jars/pom.xml]: Shall we move this to parent pom?

File: samples/webapps/jaxrs_basic/pom.xml [hunk 3]
```diff
@@ -39,13 +39,15 @@
...
                 <artifactId>maven-war-plugin</artifactId>
+                <version>${maven.war.plugin.version}</version>
                 <configuration>
                     <packagingExcludes>WEB-INF/lib/*.jar</packagingExcludes>
                     <warName>${project.artifactId}</warName>
                 </configuration>
             </plugin>
             <plugin>
                 <artifactId>maven-eclipse-plugin</artifactId>
+                <version>2.10</version>
```
Comment (chanikag) [inline samples/webapps/jaxrs_basic/pom.xml]: Shall we move this to parent pom and refer the version here

File: samples/business-process-samples/product/bpmn-sample-jars/pom.xml (hunk 2 above)
Comment (himasha) [reply samples/business-process-samples/product/bpmn-sample-jars/pom.xml]: Fixed with 
a9a4c30

File: samples/webapps/jaxrs_basic/pom.xml (hunk 3 above)
Comment (himasha) [reply samples/webapps/jaxrs_basic/pom.xml]: Fixed with 
a9a4c30

**Pull request merged**